src/components/utils/src/logger/logger_impl.cc

```diff
@@ -71,9 +73,16 @@ void LoggerImpl::DeInit() {
 }
 
 void LoggerImpl::Flush() {
-  if (use_message_loop_thread_) {
-    if (loop_thread_) {
+  if (use_message_loop_thread_ && loop_thread_) {
```

Comment:
@VladSemenyuk I would also add dcheck if `settings_` != NULL as it has a late initialization and this init call might be forgotten by developer

Reply:
assert(settings_) was added. DCHEK macro will lead to recursion because of the using logger flush fuction inside.